lib/rules/semi-style.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * @fileoverview Rule to enforce location of semicolons.
+ * @author Toru Nagashima
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const astUtils = require("../ast-utils");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const SELECTOR = `:matches(${
+    [
+        "BreakStatement", "ContinueStatement", "DebuggerStatement",
+        "EmptyStatement", "ExportAllDeclaration", "ExportDefaultDeclaration",
+        "ExportNamedDeclaration", "ExpressionStatement", "ImportDeclaration",
+        "ReturnStatement", "ThrowStatement", "VariableDeclaration"
+    ].join(",")
+})`;
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "enforce location of semicolons",
+            category: "Stylistic Issues",
+            recommended: false
+        },
+        schema: [{
+            anyOf: [
+                { enum: ["last", "first"] },
+                {
+                    type: "object",
+                    properties: {
+                        statements: { enum: ["last", "first"] },
+                        forLoopHead: { enum: ["last", "first"] }
+                    },
+                    additionalProperty: false
+                }
+            ]
+        }],
+        fixable: "whitespace"
+    },
+
+    create(context) {
+        const sourceCode = context.getSourceCode();
+        const options = context.options[0] || "last";
+        const statementLoc = typeof options === "string" ? options : (options.statements || "last");
+        const forLoopHeadLoc = typeof options === "string" ? options : (options.forLoopHead || "last");
+
+        /**
+         * Check whether comments exist between the given 2 tokens.
+         * @param {Token} left The left token to check.
+         * @param {Token} right The right token to check.
+         * @returns {boolean} `true` if comments exist between the given 2 tokens.
+         */
+        function commentsExistBetween(left, right) {
```

Comment:
Good question.
In fact, there is [here](https://github.com/eslint/eslint/pull/8101/files#diff-8129190ebfd8121cbcc8e96f0d25ae80R575) but it has not been merged. I will replace those after that is merged.